### PR TITLE
Feature/2/implement penalties

### PR DESCRIPTION
**Pull request by maticrown**

### CODE REVIEW

File: src/gui.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
+
```
Comment (EladCh) [inline src/gui.py:1]: you can remove the spare line

File: src/gui.py [hunk 2]
```diff
@@ -64,12 +67,21 @@
         bloomf_2 = BloomFilter(max_num_of_items,max_fp_rate)
     elif event == 'Insert new password':
         try:
+
```
Comment (EladCh) [inline src/gui.py:70]: you can remove the spare line

File: src/gui.py [hunk 3]
```diff
@@ -64,12 +67,21 @@
...
+            elif bloomf_1.check(values['-NEW-PASSWORD-']) or bloomf_2.check(values['-NEW-PASSWORD-']):
+                sg.PopupError("This password is already in the filter\nPlease try another password")
+            # new filter- insert
+
+            else:
+                _a, _b, _c, popup_msg, mandatory_violation = password_rules(values['-NEW-PASSWORD-'])
+                if mandatory_violation == True:
                     sg.PopupError(popup_msg)
                     continue
+
```
Comment (EladCh) [inline src/gui.py:84]: you can remove the spare line

File: src/password_strength_rules.py [hunk 4]
```diff
@@ -68,7 +75,96 @@ def password_rules(password):
             score += (len(num_of_signs)*2)
             strengths[5][1] = 'True'
 
-    # TODO penalties reduction
+
+###penalties added by me
```
Comment (EladCh) [inline src/password_strength_rules.py:79]: you can change the comment to "penalties"

File: src/password_strength_rules.py [hunk 5]
```diff
@@ -68,7 +75,96 @@ def password_rules(password):
...
+            current_string+=digit
+            sequentialDigitCount+=1
+
+        if sequentialDigitCount>1:
+            score -= sequentialDigitCount
+            popup_msg="There are sequential digits, penalty induced"
+            mandatory_violation=True
+
+    '''
+###Added by me untill here
```
Comment (EladCh) [inline src/password_strength_rules.py:167]: please remove this comment

File: src/password_strength_rules.py [hunk 6]
```diff
@@ -68,7 +75,96 @@ def password_rules(password):
...
+
+    for digit in password:
+        if prev_digit <= digit and digit.isdigit():
+            current_string+=digit
+            sequentialDigitCount+=1
+
+        if sequentialDigitCount>1:
+            score -= sequentialDigitCount
+            popup_msg="There are sequential digits, penalty induced"
+            mandatory_violation=True
```
Comment (EladCh) [inline src/password_strength_rules.py:164]: doesn't it work? if it does please change the comment. if it doesn't- do you know how to fix it? do you need help?

Comment (maticrown) [reply src/password_strength_rules.py:164]: These last two functions did not work properly when I wrote them last week and they need to be fixed

**Pull request closed without merging**